File: src/main/java/edu/cornell/cs/apl/viaduct/imp/ast/IntegerType.java
package edu.cornell.cs.apl.viaduct.imp.ast;

public final class IntegerType implements ImpType {
  private static final IntegerType INSTANCE = new IntegerType();

  private IntegerType() {}

  public static IntegerType create() {
    return INSTANCE;
  }

  @Override
  public String toString() {
    return "int";
  }
}

File: src/main/kotlin/edu/cornell/cs/apl/viaduct/syntax/types/IntegerType.kt
package edu.cornell.cs.apl.viaduct.syntax.types

import edu.cornell.cs.apl.prettyprinting.Document
import edu.cornell.cs.apl.prettyprinting.styled
import edu.cornell.cs.apl.viaduct.syntax.values.IntegerValue
import edu.cornell.cs.apl.viaduct.syntax.values.Value

/** The type of integers. */
object IntegerType : ValueType {
    override val defaultValue: Value
        get() = IntegerValue(0)

    override val asDocument: Document =
        Document("int").styled(ValueTypeStyle)
}
